jekyll/_cci2/openid-connect-tokens.adoc

```diff
@@ -81,3 +81,81 @@ The OpenID Connect ID token also contains some https://openid.net/specs/openid-c
+
+In your job, use the OpenID Connect token to do AWS STS's https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html[AssumeRoleWithWebIdentity]. Have the following information ready:
+
+* AWS region that you want to operate in
+* ARN of the IAM role that you created earlier
+
+Here is an example config that uses the AWS CLI's https://docs.aws.amazon.com/cli/latest/reference/sts/assume-role-with-web-identity.html[assume-role-with-web-identity subcommand] to authenticate with AWS. It then performs a trivial interaction (`aws sts get-caller-identity`) with AWS, demonstrating that the authentication worked. Replace that demonstration with whatever you want, such as uploading to an S3 bucket, pushing to ECR, or interacting with EKS.
+
+```yml
+jobs:
+  deploy:
+    docker:
+      - image: amazon/aws-cli
+    environment:
+      AWS_DEFAULT_REGION: <your AWS region here>
+      AWS_ROLE_ARN: <your role ARN here>
+    steps:
+      - run:
+        name: authenticate-and-interact
+        command: |
+          # use the OpenID Connect token to obtain AWS credentials
+          read -r AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN \<<< \
+            $(aws sts assume-role-with-web-identity \
+             --role-arn ${AWS_ROLE_ARN} \
+             --role-session-name "CircleCI-${CIRCLE_WORKFLOW_ID}-${CIRCLE_JOB}" \
+             --web-identity-token $CIRCLE_OIDC_TOKEN \
+             --duration-seconds 3600 \
+             --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' \
+             --output text)
+          export AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN
+          # interact with AWS
+          aws sts get-caller-identity
+```
+
+=== Advanced Usage
+
+You can take advantage of the format of the claims in CircleCI's <<format-of-the-openid-connect-id-token,OIDC token>> to limit what your CircleCI jobs can do in AWS. For example, if certain projects should only be able to access certain AWS resources, you can restrict your IAM role so that only CircleCI jobs in a specific project can assume that role.
+
+To do this, edit your IAM role's trust policy so that only an OIDC token from your chosen project can assume that role. The trust policy determines under what conditions the role can be assumed.
+
+To do this, go to an individual project's page on https://app.circleci.com/[CircleCI web app] and navigate to **Project Settings > Overview** to find your Project ID.
+
+Next, add the following condition to your role's trust policy, so that only jobs in your chosen project can assume that role. Enter your Organization ID for `<organization-id>` and your Project ID for `<project-id>`.
+
+```yml
```

Comment:
"yaml" or "yml"?

Reply:
It does not make a difference, both are accepted in terms of formatting. We tend use .yml more universally as an extension, so that's what I tend to use everywhere.

Reply:
this should be json because an IAM role's trust policy is expressed in JSON.